### PR TITLE
[CPDNQT-2152] Resend outcome to api

### PR DESCRIPTION
**Pull request by jebw**
### Context

Ticket: [CPDNPQ-2152](https://dfedigital.atlassian.net/browse/CPDNPQ-2152)

We want to allow admin users to reschedule delivery of outcomes to the qualified teachers API

### Changes proposed in this pull request

* Added logic for whether to allow rescheduling to the ParticipantOutcome model
* Added a method to `nil` out the relevant ParticipantOutcome attributes, which will trigger rescheduling
* Extended the ParticipantOutcomes component table to add a resend link
* Shortened some of the other columns in ParticipantOutcomes table to make space for the `Resend` link
* Added feature specs covering the above
* Refactored the existing method to determine whether an Outcome is the most recent to work with relationships instead of doing a query directly, this allows it to benefit from eager loaded data. The number of outcomes per declaration should be low so this shouldn't result in significant data being loaded for the `applications#show` screen.
* Extended the OutcomesTableComponent to include caption_text described the latest outcome
* Swapped the OutcomesTableComponent to store strings in translations yaml file

### Screenshots

image

[CPDNPQ-2144]: https://dfedigital.atlassian.net/browse/CPDNPQ-2144?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[CPDNPQ-2152]: https://dfedigital.atlassian.net/browse/CPDNPQ-2152?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -234,6 +234,10 @@
         resources :courses, only: %i[index show]
         resources :users, only: %i[index show]
 
+        resources :participant_outcomes, only: %i[] do
+          member { get :resend }
```
Comment (jebw) [inline config/routes.rb:238]: @cwrw I've followed ECF and made this a `GET` but it feels like this should be a `POST` since its modifying data? Thoughts?

File: spec/models/participant_outcome_spec.rb [hunk 2]
```diff
@@ -238,4 +238,71 @@
...
+      it { is_expected.to be true }
+    end
+
+    context "with incomplete delivery" do
+      let(:outcome) { create(:participant_outcome, :sent_to_qualified_teachers_api) }
+
+      it { is_expected.to be false }
+    end
+
+    context "with undelivered" do
```
Comment (cwrw) [inline spec/models/participant_outcome_spec.rb:302]: can we add a test that it resets the fields to `nil`?

File: spec/models/participant_outcome_spec.rb [hunk 3]
```diff
@@ -238,4 +238,71 @@
...
+  describe "#resend_to_qualified_teachers_api!" do
+    subject { outcome.resend_to_qualified_teachers_api! }
+
+    context "with successfully delivery" do
+      let(:outcome) { create(:participant_outcome, :successfully_sent_to_qualified_teachers_api) }
+
+      it { is_expected.to be false }
+    end
+
+    context "with failed delivery" do
```
Comment (cwrw) [inline spec/models/participant_outcome_spec.rb:290]: can we add a test that it resets the fields to `nil`?

File: config/routes.rb (hunk 1 above)
Comment (cwrw) [reply config/routes.rb:238]: yea `POST` sounds better tbh 👍

File: app/components/npq_separation/admin/outcomes_table_component.rb [hunk 4]
```diff
@@ -19,7 +19,8 @@ def head
           "Completion date",
           "Submitted by provider",
           "Sent to TRA API",
-          "Recorded by TRA API",
+          "Recorded by API",
+          "",
```
Comment (cwrw) [inline app/components/npq_separation/admin/outcomes_table_component.rb:23]: there's some `caption_text` logic in ECF, not sure if we want to bring across, don't think it's a must but just incase that was missed

**Review by cwrw — COMMENTED** [review]
looks good just some minor comments

**Review by cwrw — APPROVED** [review]
🙌
